.orchestra/support/js-prerequisites/npm

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+set -euo pipefail
+
+PORT_FILE=$(mktemp)
+CACHE_FOLDER=$(mktemp -d)
+REPO_PID=0
+SCRIPT_FOLDER="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+
+cleanup() {
+    trap - SIGINT SIGTERM ERR EXIT
+    if [[ $REPO_PID -gt 0 && -d "/proc/${REPO_PID}" ]]; then
+        kill $REPO_PID
+    fi
+    rm -rf -- "$CACHE_FOLDER"
+}
+trap cleanup SIGINT SIGTERM ERR EXIT
+
+"$SCRIPT_FOLDER/../libexec/static-npm-registry" "$NODE_CACHE" "$PORT_FILE" &
+REPO_PID=$!
+
+NEW_PATH=""
+
+while read -r COMPONENT; do
+    if [[ -n "$COMPONENT" && "$COMPONENT" != "$SCRIPT_FOLDER" ]]; then
+        NEW_PATH="$COMPONENT:$NEW_PATH"
+    fi
+done < <(echo "$PATH" | tr ':' '\n')
+
+PORT=0
+# shellcheck disable=SC2034
+for i in {1..50}; do
+    if [ ! -f "$PORT_FILE" ]; then
+        break
+    fi
+    if [ -n "$(cat "$PORT_FILE")" ]; then
+        PORT=$(cat "$PORT_FILE")
+        break
+    fi
+    sleep 0.2s
+done
+
+if [[ "$PORT" -eq 0 || ( ! -f "$PORT_FILE" ) ]]; then
+    echo "Registry daemon exited unexpectedly" > /dev/stderr
+    exit 1
+fi
+PATH="$NEW_PATH" npm --registry="http://127.0.0.1:$PORT" --cache="$CACHE_FOLDER" "$@"
```

Comment:
Mmmh, let's not tamper with `PATH`, let's just have a `npm.real` in the same directory.
The check above is rather fragile, we'd need at least to `realpath` and make sure there are no trailing slash differences..

Reply:
As discussed since `npm` is not in orchestra I've used `which -a` to determine the real npm to use